makerom/tik.c

```diff
@@ -65,7 +65,7 @@ void SetupTicketHeader(tik_hdr *hdr, cia_settings *ciaset)
 	SetLimits(hdr,ciaset);
 	
 	// Crypt TitleKey
-	if(ciaset->content.encryptCia)
+	if(ciaset->content.encryptCia || ciaset->common.titleKey)
```

Comment:
`ciaset->common.titleKey` will always evaluate to `true` because it's a defined as part of `struct cia_settings`:
`u8 titleKey[16];`

You need to respect `ciaset->content.encryptCia` because the common key may not be available in all target configurations (namely `test`).

I know `content.encryptCia` will be false by default unless otherwise specified in the RSF file. If you want to override this to true when specifying a title key, then I would do that in here:
```
cia.c

int GetSettingsFromUsrset(cia_settings *ciaset, user_settings *usrset)
{
	//...
	ciaset->content.encryptCia = usrset->common.rsfSet.Option.EnableCrypt || usrset->cia.titleKey != NULL;
	//...
}
```
Because a few lines later the ability to encrypt is checked, and turned off if not available. So it's safer this way.
```
	if(ciaset->keys->aes.commonKey[ciaset->keys->aes.currentCommonKey] == NULL && ciaset->content.encryptCia){
		fprintf(stderr,"[CIA WARNING] Common Key could not be loaded, CIA will not be encrypted\n");
		ciaset->content.encryptCia = false;
	}
```